app/Http/Requests/UpdateProviderPlatformRequest.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace App\Http\Requests;
+
+use App\Enums\ProviderPlatformState;
+use App\Enums\ProviderPlatformType;
+use Illuminate\Foundation\Http\FormRequest;
+use Illuminate\Validation\Rule;
+
+class UpdateProviderPlatformRequest extends FormRequest
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     */
+    public function authorize(): bool
+    {
+        //TODO: FIX THIS
+        return true;
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array<string, \Illuminate\Contracts\Validation\ValidationRule|array<mixed>|string>
+     */
+    public function rules(): array
+    {
+        return [
+            'name' => 'required|string|max:255',
+            'type' => [Rule::enum(ProviderPlatformType::class)],
+            'description' => 'nullable|string|max:255',
+            'icon_url' => 'required|url:http,https',
```

Comment:
For an Update on the provider platform request, I believe we should treat these as PATCH requests.

Since it requires an ID pulled from the uri to make the request in the first place(plus eventually `authorize`), I think we should make every field nullable, so the frontend can only send a JSON object with the 1-2 fields they wish to update, which will come from the Form. It will avoiding having to fetch or include unnecessary info in the request